documentation/src/docs/clients-and-streams.md

```diff
@@ -55,51 +69,120 @@ ClientConfig clientConfig = ClientConfig.builder()
 
 We can read back these events using a reader. Readers are associated with reader groups, which track the readers’ progress and allow more than one reader to coordinate over which segments they’ll read.
 
-```java
-ReaderGroupManager readerGroupManager = ReaderGroupManager
+=== "Java"
+
+    ``` java
+    ReaderGroupManager readerGroupManager = ReaderGroupManager
         .withScope("tutorial", clientConfig);
-ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+    ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
         .stream("tutorial/numbers").build();
-readerGroupManager.createReaderGroup("numReader", readerGroupConfig);
-```
+    readerGroupManager.createReaderGroup("numReader", readerGroupConfig);
+    ```
+
+=== "Python"
+
+    ``` python
+    # create a ReaderGroup rg1 against an already created Pravega scope and Stream.
+    reader_group = stream_manager.create_reader_group("numReader", "tutorial", "numbers")            
+    ```
 
 Here we used the [`ReaderGroupManager`](https://pravega.io/docs/latest/javadoc/clients/io/pravega/client/admin/ReaderGroupManager.html) to create a new reader group on numbers called `numReader`.
 
+Create a method which will be responsible for processing the event after reading :
+
+=== "Java"
+
+    ``` java
+    private void processEvent(Integer event) {
+        //Here we can write our logic how we want to act after successfully reading the event. 
+        //In this example, we are simply printing the event.
+        if (event != null) {
+            System.out.println(event);
+        }
+    }
+    ```
+
+=== "Python"
+    
+    ``` python
+    def process_event(event):
+        # Here we can write our logic how we want to act after successfully reading the event.
+        # In this example, we are simply printing the event.
+        print(event)
+    ```
+
 Now we can attach a single [`EventStreamReader`](https://pravega.io/docs/latest/javadoc/clients/io/pravega/client/stream/EventStreamReader.html) instance to our reader group and read our 3 numbers from our numbers stream:
 
-```java
-EventStreamReader<Integer> reader = factory
+=== "Java"
+
+    ``` java
+    EventStreamReader<Integer> reader = factory
         .createReader("myId", "numReader",
                       new JavaSerializer<Integer>(), ReaderConfig.builder().build());
 
-Integer intEvent;
-while ((intEvent = reader.readNextEvent(1000).getEvent()) != null) {
-    System.out.println(intEvent);
-}
+    EventRead<Integer> eventRead;
+    Integer intEvent = null;
+    while ((eventRead = reader.readNextEvent(1000)).isCheckpoint() || (intEvent = eventRead.getEvent()) != null) {
```

Comment:
It's not generally recommended to perform assignments inside of loop conditionals as it makes them look like comparisons. 
Also we should be using `isReadCompleted()` explicitly rather than simply assuming that if no event was returned and it was not because of a checkpoint that the end of the stream was reached.
So we can move `intEvent` to be inside of the loop and do a simple `while (!eventRead.isReadCompleted())`

Reply:
Done.